omniscidb/IR/Node.cpp

```diff
@@ -75,6 +75,12 @@ Node::Node(NodeInputs inputs)
     , context_data_(nullptr)
     , is_nop_(false) {}
 
+Node::Node(const Node& other)
+    : inputs_(other.inputs_)
+    , id_(crt_id_.fetch_add(1))
+    , context_data_(nullptr)
```

Comment:
So this is basically a shallow copy. Do we need it at all? I.e. would it be reasonable to have the copy constructor do a deep copy?